docs/develop/using-ai.md

```diff
@@ -0,0 +1,130 @@
+---
+title: Using AI
+sidebar_label: Using AI
+---
+
+# Using AI
+
+Creating new tables for Tailpipe plugins with AI tools and IDEs works remarkably well. At Turbot, we develop plugin tables frequently and use AI for almost every new table we create. We've experimented with various approaches, including detailed prompt engineering, explicit guidelines, IDE rules and instructions, and complex workflows, but found that AI typically produces excellent results even without heavy guidance.
+
+The key to this success is working within existing plugin repositories and opening the entire repository as a folder or project in your IDE. This gives AI tools access to existing table implementations, documentation examples, code patterns, and naming conventions to generate consistent, high-quality results without extensive prompting.
+
+If you're looking to use AI to query Tailpipe rather than develop new tables, you can use the [Tailpipe MCP server](https://github.com/turbot/tailpipe-mcp), which provides powerful tools for AI agents to inspect tables and run queries.
+
+## Getting Started
+
+While AI often works well with simple requests like "Create a table for [log_type]", here are some prompts we use at Turbot that you may find helpful as starting points.
+
+### Prerequisites
+
+1. Open the plugin repository in your IDE (Cursor, VS Code, Windsurf, etc.) to give AI tools access to all existing code and documentation.
+2. Ensure you have Tailpipe installed (`brew install turbot/tap/tailpipe` for MacOS or the installation script for Linux/WSL).
+3. Set up access credentials for the cloud provider (e.g., AWS credentials).
+4. Configure test log sources (e.g., S3 buckets with sample logs, CloudWatch Log Groups).
+5. Configure the [Tailpipe MCP server](https://github.com/turbot/tailpipe-mcp) which allows the agent to inspect tables and run queries.
+
+### Create Table
+
+First, create the new table and its documentation, using existing tables and docs as reference.
+
+```md
+Your goal is to create a new Tailpipe table and documentation for <log_type>.
+
+1. Review existing tables and their documentation in the plugin to understand:
+  - Table structure patterns and naming conventions
+  - Source configurations (S3, CloudWatch, etc.)
+  - Standard field enrichment patterns
+  - Column structures and data types
+
+2. Create the table implementation with:
+  - Proper source metadata configuration
+  - Row enrichment logic for standard and log-specific fields
+  - Extractor implementation for parsing logs
```

Comment:
`Extractor` isn't required 99% of the time, usually a `Mapper` is all that is required.